neard/src/runtime.rs

```diff
@@ -210,6 +210,27 @@ impl NightshadeRuntime {
         (self.store.clone(), store_update, state_roots)
     }
 
+    fn get_epoch_validators(
+        &self,
+        epoch_id: &EpochId,
+        last_block_hash: &CryptoHash,
+    ) -> Result<HashMap<AccountId, Balance>, Error> {
+        let mut epoch_manager = self.epoch_manager.write().expect(POISONED_LOCK_ERR);
```

Comment:
Do you use `.write()` to have exclusive atomic access to the epoch manager?

Reply:
No. `get_slashed_validators` and `get_epoch_info` are both mutable because of the cache we maintain internally.